dbms/src/IO/readDecimalText.h

```diff
@@ -145,10 +132,17 @@ inline bool tryReadDecimalText(ReadBuffer & buf, T & x, unsigned int precision,
     unsigned int digits = precision;
     int exponent;
 
-    if (!readDigits<false>(buf, x, digits, exponent, true) ||
-        static_cast<int>(digits) + exponent > static_cast<int>(precision - scale) ||
-        static_cast<int>(scale) + exponent < 0)
+    try
+    {
+        readDigits(buf, x, digits, exponent, true);
+        if (static_cast<int>(digits) + exponent > static_cast<int>(precision - scale) ||
+            static_cast<int>(scale) + exponent < 0)
+            return false;
+    }
+    catch (const Exception &)
```

Comment:
This is extremely slow in exceptional case.
Not viable solution to process column data.